client/keys.go

```diff
@@ -91,16 +91,15 @@ func runAddCmd(cmd *cobra.Command, args []string) error {
 	)
 
 	dryRun, _ := cmd.Flags().GetBool(flags.FlagDryRun)
+	newcontext := clientCtx
 	if dryRun {
 		kr, err = keyring.New(sdk.KeyringServiceName(), keyring.BackendMemory, clientCtx.KeyringDir, buf, hd.EthSecp256k1Option())
-	} else {
-		backend, _ := cmd.Flags().GetString(flags.FlagKeyringBackend)
-		kr, err = keyring.New(sdk.KeyringServiceName(), backend, clientCtx.KeyringDir, buf, hd.EthSecp256k1Option())
+		newcontext = clientCtx.WithKeyring(kr)
```

Comment:
This only overrides the keyring 👍

```suggestion
		clientCtx = clientCtx.WithKeyring(kr)
```